editoast/src/views/v2/pathfinding.rs

```diff
@@ -0,0 +1,366 @@
+    /// Empty if does not support any electrification
+    rolling_stock_supported_electrification: Vec<String>,
+    /// List of supported signaling systems
+    rolling_stock_supported_signaling_systems: Vec<String>,
+    /// List of waypoints given to the pathfinding
+    path_items: Vec<PathItemLocation>,
+}
+
+/// Compute a pathfinding
+#[utoipa::path(
+    tag = "pathfindingv2",
+    request_body = PathfindingInput,
+    responses(
+        (status = 200, description = "Pathfinding Result", body = PathfindingResult),
+    ),
+)]
+#[post("")]
+async fn post(
+    db_pool: Data<DbPool>,
+    redis_client: Data<RedisClient>,
+    infra_id: Path<i64>,
+    data: Json<PathfindingInput>,
+) -> Result<Json<PathfindingResult>> {
+    let infra_id = infra_id.into_inner();
+    let path_input = data.into_inner();
+    let conn = &mut db_pool.get().await?;
+    let mut redis_conn = redis_client.get_connection().await?;
+    Ok(Json(
+        pathfinding_blocks(conn, &mut redis_conn, infra_id, &path_input).await?,
+    ))
+}
+
+pub async fn pathfinding_blocks(
+    conn: &mut PgConnection,
+    redis_conn: &mut RedisConnection,
+    infra_id: i64,
+    path_input: &PathfindingInput,
+) -> Result<PathfindingResult> {
+    use crate::models::Infra;
+    use crate::models::Retrieve;
+    // Deduce infra version from infra
+    let infra = Infra::retrieve_conn(conn, infra_id)
+        .await?
+        .ok_or(PathfindingError::InfraNotFound { infra_id })?;
+    let infra_version = infra.version.unwrap();
+    // Compute unique hash of PathInput
+    let hash = path_input_hash(infra_id, &infra_version, path_input);
+    // Try to retrieve the result from Redis
+    let result: Option<PathfindingResult> = redis_conn.json_get(&hash).await?;
+    if let Some(pathfinding) = result {
+        return Ok(pathfinding);
+    }
+    // If miss cache:
+    // 1) extract locations from path items
+    let path_items = path_input.clone().path_items;
+    let result = extract_location_from_path_items(conn, infra_id, &path_items).await?;
+    match result {
+        Ok(track_offsets) => {
+            for tracks in track_offsets {
+                let ids = tracks
+                    .iter()
+                    .map(|track_offset| (infra_id, track_offset.track.0.clone()));
+                let (_, missings): (Vec<_>, _) =
+                    TrackSectionModel::retrieve_batch(conn, ids).await?;
+                if !missings.is_empty() {
+                    return Ok(PathfindingResult::NotFoundInTracks);
+                }
+            }
+        }
+        Err(e) => return Ok(e),
+    };
+    // 2) compute path from core
+    let result = extract_location_from_path_items(conn, infra_id, &path_items).await?;
+    let res = match result {
+        Ok(track_offsets) => pathfinding_core(track_offsets),
+        Err(e) => return Ok(e),
+    };
+    let res = res.unwrap();
+    // 3) Put in cache
+    let seconds = chrono::TimeDelta::weeks(1).num_seconds();
```

Comment:
We can `use chrono::Duration` in headers.

```suggestion
    let seconds = Duration::weeks(1).num_seconds();
```